src/aws/osml/model_runner/app.py

```diff
@@ -916,3 +912,18 @@ def get_extents(ds: gdal.Dataset) -> dict[str, Any]:
 
         extents = {"north": maxy, "south": miny, "east": maxx, "west": minx}
         return extents
+
+    @staticmethod
+    def calculate_region_status(total_tile_count: int, tile_error_count: int) -> RegionRequestStatus:
+        """
+        Calculate the processing status of a region upon completion
+        :param total_tile_count: number of tiles that were processed
+        :param tile_error_count: number of tiles with errors
+        :return: RegionRequestStatus
+        """
+        region_status = RegionRequestStatus.SUCCESS
+        if total_tile_count == tile_error_count:
+            region_status = RegionRequestStatus.FAILED
+        if tile_error_count > 0 and tile_error_count > total_tile_count:
```

Comment:
The second half of this if statement looks broken. When would the tile_error_count ever be greater than the total_tile_count? Wouldn't this function default to SUCCESS for a failure count > 0 but less than the total_tile_count?

Reply:
yeah, the last line of that should be `if tile_error_count > 0 and tile_error_count < total_tile_count.`  I'll fix that in the unit test PR.  I was planning to add a test specifically for his function anyway.